src/components/pager/pager-navigation.component.js

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withTheme } from 'styled-components';
+import I18n from 'i18n-js';
+import {
+  PagerNavigationStyles,
+  PagerNavInnerStyles,
+  PagerNoSelectStyles,
+  PagerButtonWrapperStyles,
+  PagerLinkStyles
+} from './pager.styles';
+import Icon from '../icon';
+import NumberComponent from '../number';
+import NumberInput from '../../__experimental__/components/number';
+import Events from '../../utils/helpers/events';
+import baseTheme from '../../style/themes/base';
+import { isClassic } from '../../utils/helpers/style-helper';
+
+const PagerNavigation = (props) => {
+  const { theme, setCurrentThemeName } = props;
+  setCurrentThemeName(theme.name);
+
+  const updatePageFromInput = (ev) => {
+    let newPage = Math.abs(Number(ev.target.value));
+
+    if (Number(newPage) === 0 || Number.isNaN(newPage)) newPage = '1';
+    else if (newPage > maxPages()) newPage = String(maxPages());
+
+    props.onPagination(String(newPage), props.pageSize, 'input');
+  };
+
+  function maxPages() {
+    const totalRecordsCount = props.totalRecords >= 0 ? props.totalRecords : 0;
+
+    if (props.pageSize && props.pageSize !== '0' && totalRecordsCount > 0) {
+      return Math.ceil(totalRecordsCount / props.pageSize);
+    }
+    return 1;
+  }
+
+  function navArrowChange(step) {
+    const newPage = String(Number(props.currentPage) + step);
+    const desc = step === 1 ? 'next' : 'previous';
+    props.onPagination(newPage, props.pageSize, desc);
+  }
+
+  function navArrow(step) {
+    const arrowProps = {
+      'data-element': 'next-page',
+      type: 'dropdown'
+    };
+
+    function disabled() {
+      if (step === -1) {
+        return props.currentPage === '1';
+      }
+      return props.currentPage === String(maxPages());
+    }
+
+    return (
+      <PagerButtonWrapperStyles disabled={ disabled() } next={ step === 1 }>
+        <Icon
+          onClick={ () => {
+            if (!disabled()) { navArrowChange(step); }
+          } }
+          { ...arrowProps }
+        />
+      </PagerButtonWrapperStyles>
+    );
+  }
+
+  function navLink(type) {
+    const currentPage = Number(props.currentPage);
+    const navLinkConfig = {
+      first: {
+        text: 'First',
+        destination: '1'
+      },
+      last: {
+        text: 'Last',
+        destination: String(maxPages())
+      },
+      next: {
+        text: 'Next',
+        destination: String(currentPage + 1)
+      },
+      back: {
+        text: 'Previous',
+        destination: String(currentPage - 1)
+      }
+    };
+
+    function disabled() {
+      if (currentPage === 1) {
+        return type === 'back' || type === 'first';
+      }
+      if (currentPage === maxPages()) {
+        return type === 'next' || type === 'last';
+      }
+      return false;
+    }
+
+    const pagerLinkProps = {
```

Comment:
Just wondering if lines 103-113 could be abbreviated a bit? Something like this could make things a bit shorter, and you could then use `disabled`, rather than `isDisabled` in the styled component - at the moment it looks a bit odd that we have both `disabled` and `isDisabled` as props on the same object.
```
const clickHandler = () => props.onPagination(navLinkConfig[type].destination, props.pageSize, type);

const pagerLinkProps = {
  disabled: disabled(),
  onClick: disabled() ? clickHandler: undefined
};
```